src/commands/Minion/zalcano.ts

```diff
@@ -0,0 +1,104 @@
+import { calcWhatPercent, percentChance, reduceNumByPercent } from 'e';
+import { CommandStore, KlasaMessage } from 'klasa';
+
+import { BotCommand } from '../../lib/BotCommand';
+import { Activity, Tasks, Time } from '../../lib/constants';
+import { MinigameIDsEnum } from '../../lib/minions/data/minigames';
+import { minionNotBusy, requiresMinion } from '../../lib/minions/decorators';
+import removeFoodFromUser from '../../lib/minions/functions/removeFoodFromUser';
+import { UserSettings } from '../../lib/settings/types/UserSettings';
+import { SkillsEnum } from '../../lib/skilling/types';
+import { ZalcanoActivityTaskOptions } from '../../lib/types/minions';
+import { formatDuration } from '../../lib/util';
+import addSubTaskToActivityTask from '../../lib/util/addSubTaskToActivityTask';
+
+export default class extends BotCommand {
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			oneAtTime: true,
+			altProtection: true,
+			requiredPermissions: ['ADD_REACTIONS', 'ATTACH_FILES']
+		});
+	}
+
+	calcPerformance(kcLearned: number, skillPercentage: number) {
+		let basePerformance = 50;
+
+		// Up to +25% performance for KC
+		basePerformance += Math.floor(kcLearned / 4);
+
+		// Up to +20% Performance for Skill Levels
+		basePerformance += Math.floor(skillPercentage / 10);
+
+		return Math.min(100, basePerformance);
+	}
+
+	@minionNotBusy
+	@requiresMinion
+	async run(msg: KlasaMessage) {
+		if (
+			msg.author.skillLevel(SkillsEnum.Mining) < 70 ||
+			msg.author.skillLevel(SkillsEnum.Smithing) < 70 ||
+			msg.author.settings.get(UserSettings.QP) < 150
+		) {
+			return msg.send(
+				`To fight Zalcano, you need: Level 70 Mining, Level 70 Smithing and 150 QP.`
+			);
+		}
+
+		const kc = msg.author.getMinigameScore(MinigameIDsEnum.Zalcano);
+		const kcLearned = Math.min(100, calcWhatPercent(kc, 100));
+
+		const boosts = [];
+		let baseTime = Time.Minute * 6;
+		baseTime = reduceNumByPercent(baseTime, kcLearned / 6);
+		boosts.push(`${(kcLearned / 6).toFixed(2)}% boost for experience`);
+
+		const skillPercentage =
+			msg.author.skillLevel(SkillsEnum.Mining) + msg.author.skillLevel(SkillsEnum.Smithing);
```

Comment:
After looking some YouTube videos where they did comparison between a 70 mining account and 96 mining account. There is some slight difference, but it only scales with mining and not smithing. So removing the smithing part makes sense. Other than that the rest of the code looks good, maybe have some kind of small scaling with the pickaxes but only like 1,2,3,4,5%

Also either having graceful gears as a requirement or boost makes sense since everyone on the youtube videos uses is.